themes/default/content/blog/pulumi-targets-dotnet-6/index.md

```diff
@@ -0,0 +1,182 @@
+
+```cs
+return await Deployment.RunAsync(() =>
+{
+   var bucket = new Bucket("my-bucket");
+});
+```
+
+Moreover, when you want to `await` asynchronous functions alongside your resource definitions, just mark the lambda as `async`. Here is an example where we await function invokes and work with their results:
+
+```cs
+using Pulumi;
+using Pulumi.Aws;
+using Pulumi.Aws.Ec2;
+using System.Linq;
+
+return await Deployment.RunAsync(async() =>
+{
+    var vpc = await GetVpc.InvokeAsync(new()
+    {
+        Default = true  
+    });
+
+    var zones = await GetAvailabilityZones.InvokeAsync();
+
+    foreach (var range in zones.Names.Select((zone, index) => new { Value = zone, Key = index }))
+    {
+        var subent = new Subnet($"vpcSubnet-{range.Key}", new()
+        {
+            AssignIpv6AddressOnCreation = false,
+            VpcId = vpc.Id,
+            MapPublicIpOnLaunch = true,
+            CidrBlock = $"10.100.{range.Key}.0/24",
+            AvailabilityZone = range.Value,
+            Tags =
+            {
+               ["Name"] = $"pulumi-sn-{range.Value}",
+            },
+        });
+    }
+});
+```
+
+The cherry on top is that you can use [target-typed `new` expressions](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/target-typed-new) when instantiating resource or function invoke argument types. Basically instead of writing `new Bucket("my-bucket", new BucketArgs { ... })`, you can write `new Bucket("my-bucket", new () { ... })`.
+
+### Backward-compatibility
+
+It is worth mentioning that although the new templates and the accompanying getting-started guides are updating to .NET 6, the main `Pulumi` SDK NuGet package and cloud provider packages are still targeting `netcoreapp3.1`. This mean that if users are still using .NET SDK v3.1 or v5.0 to build and run your projects, everything will continue to work as always.
+
+### Code generation
+
+Pulumi provides tools to translate from several IaC languages and generates Pulumi programs in any of our supported languages. This includes:
+
+- Translating [Azure Resource Manager (ARM)](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/overview) templates using[arm2pulumi](https://www.pulumi.com/arm2pulumi/)
+- Translating Terrafrom code using [tf2pulumi](https://www.pulumi.com/tf2pulumi/)
+- Translating Kubernetes YAML using [kube2pulumi](https://www.pulumi.com/kube2pulumi/)
+- Translating [Pulumi YAML](https://www.pulumi.com/docs/intro/languages/yaml/) with the Pulumi CLI using `pulumi convert`
+
+All of these tools take their input language, convert it into the Pulumi intermediate language, then translate it into one of our supported languages. In case of C#, the code generation has been updated to generate code that uses latest features from .NET 6. It also fixed small issues that were present in the translation logic.
```

Comment:
Hmm, this first clause feels a little colloquial to me, but I'm not sure how I'd modify it. Here's one suggestion.

```suggestion
These tools take their source language as input, convert it into the Pulumi intermediate language, then translate it into one of our supported languages. In the case of C#, the code generation has been updated to generate code that uses the latest features from .NET 6. It also fixed minor issues that were present in the translation logic.
```